File: leetcode/go/strings/reverse_string_test.go
package reverse_string_test

import (
	"bytes"
	"testing"
)

// https://leetcode.com/explore/interview/card/top-interview-questions-easy/127/strings/879/
func reverseString(s []byte) {
	right := len(s) - 1

	for left := 0; left < right; left++ {
		temp := s[left]

		s[left] = s[right]
		s[right] = temp

		right--
	}
}

func TestReverseString(t *testing.T) {
	expectedBytes := []byte("olleh")
	str := []byte{'h', 'e', 'l', 'l', 'o'}
	reverseString(str)

	if !bytes.Equal(str, expectedBytes) {
		t.Errorf("expected %v but got %v", expectedBytes, str)
	}
}

File: leetcode/typescript/src/strings/reverseString.test.ts
import { describe, expect, test } from 'bun:test'

// https://leetcode.com/explore/interview/card/top-interview-questions-easy/127/strings/879/
function reverseString(s: string[]): void {
  let left = 0
  let right = s.length - 1

  while (left < right) {
    let temp = s[left]
    s[left] = s[right]
    s[right] = temp

    left++
    right--
  }
};

describe("strings: reverseString", () => {
  test("reverses string in place", () => {
    let str = ['h', 'e', 'l', 'l', 'o']
    reverseString(str)
    expect(str).toStrictEqual(['o', 'l', 'l', 'e', 'h'])
  });
});
